lib/libp2p_port.ex

```diff
@@ -81,18 +168,62 @@ defmodule LambdaEthereumConsensus.Libp2pPort do
   ### PRIVATE FUNCTIONS
   ######################
 
-  defp handle_notification(%Libp2pProto.Notification{
-         n: {:gossip, %Libp2pProto.GossipSub{topic: topic, message: message}}
+  defp handle_notification(%Notification{
+         n: {:gossip, %GossipSub{topic: topic, message: message}}
        }) do
     Logger.info("[Topic] #{topic}: #{message}")
   end
 
-  defp handle_notification(%Libp2pProto.Notification{
-         n: {:response, %Libp2pProto.Response{id: id, success: success, message: message}}
+  defp handle_notification(%Notification{
+         n:
+           {:request,
+            %Request{
+              protocol_id: protocol_id,
```

Comment:
I'm not sure we need the protocol_id AND the handler. Not sure we need it as a fixed pid either. We can just have handler modules per protocol_id. Also, what is a protocol_id? Do we have examples? Or a list of them? Is it size fixed or can there be multiple of them?

I want to avoid unnecessary early abstractions unless we are sure that we need them. Some documentation with examples either in the code on in markdown may help understand how we're gonna use them.

For example, if with protocol we mean "request-response", we can just have a module here.

Reply:
If this is just about "/eth2/beacon_chain/req/beacon_blocks_by_range/2/ssz_snappy" I definitely wouldn't add this level of abstraction, I would just hardcode that in the other side and have a hardcoded handler for it.

The idea behind the port is not to have a very thin layer over every libp2p call, but communicate with an app that solves a bit of the problem. We need to define the frontier represented by the port a bit better.

Reply:
The protocol ID would be the `"/eth2/beacon_chain/req/beacon_blocks_by_range/2/ssz_snappy"` string, yes. The size is fixed, and there are multiple of them. There is a test using all of the implemented API in `libp2p_port_test.exs`, but an example usage would be:

```elixir
# start port...
Libp2pPort.set_handler("/eth2/...")
{"/eth2/...", message_id, payload} = Libp2pPort.handle_request()
# process payload and generate response...
Libp2pPort.send_response(message_id, response)
```

I think hardcoding the specific protocol IDs or the handlers would make things more difficult both for testing and for general usage, but storing the handlers on Elixir would certainly simplify things. Will try that